app/pages/home-for-user/index.jsx

```diff
@@ -1,75 +1,76 @@
 import React from 'react';
 import { Link } from 'react-router';
-import BlurredImage from './blurred-image';
+import qs from 'qs';
 import Pullout from 'react-pullout';
+import BlurredImage from './blurred-image';
 import getUserRibbonData from '../../lib/get-user-ribbon-data';
 import CircleRibbon from './circle-ribbon';
 import RecentProjectsSection from './recent-projects';
 import RecentCollectionsSection from './recent-collections';
 import RecentMessagesSection from './recent-messages';
 import MyBuildsSection from './my-builds';
 import ProjectStats from './project-stats';
-import qs from 'qs';
 import HomePageSocial from '../home-not-logged-in/social';
 import NewsSection from './news-pullout';
-
+import mediaActions from '../lab/actions/media';
 
 const SECTIONS = {
   projects: RecentProjectsSection,
   collections: RecentCollectionsSection,
   messages: RecentMessagesSection,
-  builds: MyBuildsSection,
+  builds: MyBuildsSection
 };
 
-const HomePageForUser = React.createClass({
-  propTypes: {
-    user: React.PropTypes.object.isRequired,
-    location: React.PropTypes.object,
-  },
-
-  contextTypes: {
-    setAppHeaderVariant: React.PropTypes.func,
-  },
+export default class HomePageForUser extends React.Component {
+  constructor(props) {
+    super(props);
 
-  getDefaultProps() {
-    return {
-      user: {},
-    };
-  },
-
-  getInitialState() {
-    return {
+    this.state = {
+      buttonFocus: false,
       backgroundSrc: '',
       avatarSrc: '',
       showNews: false,
       ribbonData: [],
       loading: false,
       error: null,
+      pendingMedia: [],
       selectedProjectID: null,
-      openSection: null,
+      openSection: null
     };
-  },
+
+    this.toggleNews = this.toggleNews.bind(this);
+    this.toggleFocus = this.toggleFocus.bind(this);
+    this.updateBackground = this.updateBackground.bind(this);
+    this.createLinkedResource = this.props.actions.createLinkedResource.bind(this);
+    this.uploadMedia = this.props.actions.uploadMedia.bind(this);
+  }
 
   componentDidMount() {
+    const backgroundBtn = document.getElementById('change-background');
+    backgroundBtn.addEventListener('focus', this.toggleFocus, true);
+    backgroundBtn.addEventListener('blur', this.toggleFocus, true);
```

Comment:
`tabIndex` isn't needed on an input and shouldn't be used on a label (clicking on a label should focus the associated input). What you're trying to do here is style the adjacent text when the input has focus. One way to do this with just CSS is by wrapping the label text in a span and using sibling selectors `input:focus + span`.